client/components/LearnerViewDirective.js

```diff
@@ -481,6 +505,19 @@ tie.directive('learnerView', [function() {
         var autosaveCancelPromise;
         var cachedCode;
         var congratulatoryFeedback = FeedbackObjectFactory.create();
+        var reinforcement = ReinforcementObjectFactory.create();
+        QuestionDataService.initCurrentQuestionSet(questionSetId);
+        var questionSet = QuestionDataService.getCurrentQuestionSet(
+          questionSetId);
+        $scope.currentQuestionIndex = 0;
```

Comment:
Don't copy/paste please -- refactor common stuff into a function instead. That said, I just made a similar comment on #250 which has just been merged, so merging from master into this branch and resolving the conflicts should address this.